packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -0,0 +1,512 @@
+                        this.selectedImage &&
+                        this.wrapper &&
+                        this.clonedImage
+                    ) {
+                        updateWrapper(
+                            this.imageEditInfo,
+                            this.options,
+                            this.selectedImage,
+                            this.clonedImage,
+                            this.wrapper,
+                            undefined,
+                            this.croppers
+                        );
+                        this.isCropMode = true;
+                    }
+                },
+                this.zoomScale
+            ),
+        ];
+
+        updateWrapper(
+            this.imageEditInfo,
+            this.options,
+            this.selectedImage,
+            this.clonedImage,
+            this.wrapper,
+            undefined,
+            this.croppers
+        );
+    }
+
+    private editImage(
+        editor: IEditor,
+        image: HTMLImageElement,
+        apiOperation: ImageEditOperation,
+        operation: (imageEditInfo: ImageMetadataFormat) => void
+    ) {
+        if (this.wrapper && this.selectedImage && this.shadowSpan) {
+            image = this.removeImageWrapper() ?? image;
+        }
+        this.startEditing(editor, image, apiOperation);
+        if (!this.selectedImage || !this.imageEditInfo || !this.wrapper || !this.clonedImage) {
+            return;
+        }
+
+        operation(this.imageEditInfo);
+
+        updateWrapper(
+            this.imageEditInfo,
+            this.options,
+            this.selectedImage,
+            this.clonedImage,
+            this.wrapper
+        );
+
+        this.formatImageWithContentModel(
+            editor,
+            true /* shouldSelect*/,
+            true /* shouldSelectAsImageSelection*/
+        );
+    }
+
+    private cleanInfo() {
+        this.editor?.setEditorStyle('imageEdit', null);
+        this.selectedImage = null;
+        this.shadowSpan = null;
+        this.wrapper = null;
+        this.imageEditInfo = null;
+        this.imageHTMLOptions = null;
+        this.dndHelpers.forEach(helper => helper.dispose());
+        this.dndHelpers = [];
+        this.clonedImage = null;
+        this.lastSrc = null;
+        this.wasImageResized = false;
+        this.isCropMode = false;
+        this.resizers = [];
+        this.rotators = [];
+        this.croppers = [];
+    }
+
+    private formatImageWithContentModel(
+        editor: IEditor,
+        shouldSelectImage: boolean,
+        shouldSelectAsImageSelection: boolean
+    ) {
+        if (
+            this.lastSrc &&
+            this.selectedImage &&
+            this.imageEditInfo &&
+            this.clonedImage &&
+            this.shadowSpan
+        ) {
+            editor.formatContentModel(
+                (model, _) => {
+                    const selectedSegmentsAndParagraphs = getSelectedSegmentsAndParagraphs(
```

Comment:
So here we edit image based on current selection, then why we need to pass in image object in other editing functions?

Reply:
We pass the image element to add and remove the wrapper element. Here we edit the selected content model image